From 10a6bc26e2f6feb14b3b59036395d01a65a2b271 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Tue, 28 Dec 2021 17:38:45 +0100
Subject: [PATCH 2/8] setupapi: Start auto-start services on registration.

This makes sure PlugPlay service is also started after prefix
initialization. Otherwise we can also use 0x800 flag in wine.inf.in.
---
 dlls/setupapi/install.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/setupapi/install.c b/dlls/setupapi/install.c
index df39fd440b3..fd06db71181 100644
--- a/dlls/setupapi/install.c
+++ b/dlls/setupapi/install.c
@@ -1406,6 +1406,7 @@ static BOOL add_service( SC_HANDLE scm, HINF hinf, const WCHAR *name, const WCHA
     }
     iterate_section_fields( hinf, section, L"BitReg", bitreg_callback, NULL );
 
+    if (start_type < SERVICE_DEMAND_START) flags |= SPSVCINST_STARTSERVICE;
     if (flags & SPSVCINST_STARTSERVICE) StartServiceW( service, 0, NULL );
     CloseServiceHandle( service );
 
-- 
2.34.1

